src/txn/txn.c

```diff
@@ -878,6 +878,77 @@ __txn_append_hs_record(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, WT_ITEM *
     return (ret);
 }
 
+/*
+ * __txn_commit_timestamps_usage_check --
+ *     Print warning messages when encountering surprise timestamp usage.
+ */
+static inline int
+__txn_commit_timestamps_usage_check(WT_SESSION_IMPL *session, WT_TXN_OP *op, WT_UPDATE *upd)
+{
+    WT_TXN *txn;
+    wt_timestamp_t op_ts, prev_op_durable_ts;
+    uint32_t ts_flags;
+    char ts_string[2][WT_TS_INT_STRING_SIZE];
+    bool txn_has_ts;
+
+    txn = session->txn;
+    txn_has_ts = F_ISSET(txn, WT_TXN_HAS_TS_COMMIT) || F_ISSET(txn, WT_TXN_HAS_TS_DURABLE);
+
+#define WT_COMMIT_TS_VERB_PREFIX "Commit timestamp unexpected usage: "
+
+    /* Nothing to do if no tables this transaction touched were configured for verbose logging. */
```

Comment:
This sentence is awkwardly worded. Maybe something like:
If this transaction did not touch any table configured for verbose logging, we're done.

Reply:
1a9d666